sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/FluxAutoRenewTest.java

```diff
@@ -0,0 +1,431 @@
+    }
+
+    /**
+     * Test that the user can cancel the receive function.
+     */
+    @Test
+    void canCancel() {
+        // Arrange
+        final ServiceBusReceivedMessage message2 = new ServiceBusReceivedMessage("data".getBytes());
+        message2.setLockToken(UUID.randomUUID());
+        message2.setLockedUntil(OffsetDateTime.now().plusSeconds(2));
+
+        final FluxAutoRenew renewOperator = new FluxAutoRenew(messageSource,
+            MAX_AUTO_LOCK_RENEW_DURATION, messageLockContainer, renewalFunction);
+
+        // Act & Assert
+        StepVerifier.create(renewOperator)
+            .then(() -> {
+                messagesPublisher.next(message);
+                messagesPublisher.next(message2);
+            })
+            .assertNext(actual -> Assertions.assertEquals(LOCK_TOKEN_STRING, actual.getLockToken()))
+            .thenCancel()
+            .verify();
+
+    }
+
+    @Test
+    public void canMap() {
+        // Arrange
+        final String expectedMappedValue = "New Expected Mapped Value";
+        final FluxAutoRenew renewOperator = new FluxAutoRenew(messageSource,
+            MAX_AUTO_LOCK_RENEW_DURATION, messageLockContainer, renewalFunction);
+
+        // Act & Assert
+        StepVerifier.create(renewOperator.map(serviceBusReceivedMessage -> expectedMappedValue))
+            .then(() -> {
+                messagesPublisher.next(message);
+            })
+            .assertNext(actual -> Assertions.assertEquals(expectedMappedValue, actual))
+            .thenCancel()
+            .verify();
+    }
+
+    /**
+     * Check that illegal values are not allowed in constructor.
+     */
+    @Test
+    void illegalValueConstructor() {
+        // Arrange, Act & Assert
+        assertThrows(NullPointerException.class, () -> new FluxAutoRenew(null,
+            MAX_AUTO_LOCK_RENEW_DURATION, messageLockContainer, renewalFunction));
+
+        assertThrows(NullPointerException.class, () -> new FluxAutoRenew(messageSource,
+            MAX_AUTO_LOCK_RENEW_DURATION, null, renewalFunction));
+
+        assertThrows(NullPointerException.class, () -> new FluxAutoRenew(messageSource,
+            MAX_AUTO_LOCK_RENEW_DURATION, messageLockContainer, null));
+
+        assertThrows(IllegalArgumentException.class, () -> new FluxAutoRenew(messageSource,
+            DISABLE_AUTO_LOCK_RENEW_DURATION, messageLockContainer, renewalFunction));
+
+    }
+
+    /**
+     * Test that the function to renew lock is invoked. It will verify
+     * 1. The renew lock function is invoked multiple times.
+     * 2. The updated new lockedUntil is reflected on ServiceBusReceivedMessage object.
+     */
+    @Test
+    void lockRenewedMultipleTimes() {
+        // Arrange
+        final int renewedForAtLeast = 3;
+        final int totalProcessingTimeSeconds = 5;
+        final AtomicInteger actualTokenRenewCalledTimes = new AtomicInteger();
+        final Function<String, Mono<OffsetDateTime>> lockTokenRenewFunction = (lockToken) -> {
```

Comment:
If you  mock this function, you can verify invocations it rather than keeping track.

Reply:
Also, you could have mocked the function in general instead of providing a concrete implementation.

Reply:
I could not find a way to mock `onRenewLock.**andThen**(...)`, So for now I am using `concrete implementation` for renewLock.